src/hooks/stateManagement.ts

```diff
@@ -37,7 +38,23 @@ export function useArrayEdit<T, K extends string | number>(
             return newValues;
         });
     };
-    return [addItem, removeItem, modifyItem];
+    const patchItem = (modifiedItem: T) => {
+        setValues((oldValues) => {
+            const key = keySelector(modifiedItem);
+            const index = oldValues.findIndex(item => keySelector(item) === key);
+            if (index === -1) {
+                return oldValues;
+            }
+            const newValues = [...oldValues];
+            const mergedItem = {
+                ...oldValues[index],
+                ...modifiedItem,
+            };
+            newValues.splice(index, 1, mergedItem);
+            return newValues;
+        });
+    };
```

Comment:
Doesn't make sense to have a different patchItem.
We can easy accomplish this by passing `modifiedItem: T` as `modifiedItem: T | (oldVal: T) => T`